pkg/pipeline/stream/message.go

```diff
@@ -272,6 +272,19 @@ func (m *Message) Status() MessageStatus {
 	}
 }
 
+// StatusError returns the error that was returned when the message was acked or
+// nacked. If the message was successfully acked/nacked or it is still open the
+// method returns nil.
+func (m *Message) StatusError() error {
+	switch m.Status() {
+	case MessageStatusAcked:
+		return m.Ack()
```

Comment:
This is to cover the DLQ case?

Reply:
This specific line is to cover the case if a message is acknowledged by a processor (i.e. filtered out) and we fail to deliver the acknowledgment to the connector - in that case, the processor node will return an error, causing the whole pipeline to stop. The next case is for the DLQ, in case the processor failed to process the message and nacked it but the nack failed (i.e. it wasn't stored in the DLQ), that also causes the processor node to fail and stop the pipeline.

So essentially the result of this function tells us if the processor node is running or if it failed to process the message and it stopped. Based on this we can stop the goroutine that's wrapping the processor node.

Reply:
That's some useful info, it would be nice to have a short comment about that. :+1:

Reply:
But is this the right place? 🤔 I mean this function can be used in other places as well. I can try to add this info somewhere in the parallel node though.

Reply:
WDYT? https://github.com/ConduitIO/conduit/pull/744/commits/b508934b9429b575b6973c0eb42b6b7c31f9a6f2

Reply:
Excellent!